src/loaders/common/loader_helper.py

```diff
@@ -77,3 +84,53 @@ def init_genome_atrri_doc(kbase_collection, load_version, genome_id):
     }
 
     return doc
+
+
+def get_token(token_filename):
+    """
+    Get token if a filename is provided at home dir.
+    """
+    try:
+        home_dir = os.path.expanduser("~")
+        with open(os.path.join(home_dir, "." + token_filename), "r") as f:
+            token = f.readline().strip()
+    except FileNotFoundError as e:
+        raise e
+    return token
+
+
+def start_podman_service(uid):
+    """
+    Start podman service. Used by workspace_downloader.py script.
+    """
+    # TODO find out the right way to check if a podman service is running
+    proc = subprocess.Popen(["podman", "system", "service", "-t", "0"])
+    os.environ["DOCKER_HOST"] = DOCKER_HOST.format(uid)
+    return proc
+
+
+def is_upa_info_complete(output_dir, upa):
+    """
+    Check whether an UPA needs to be downloaded or not by loading the metadata file.
+    Make sure it has all the right keys.
+    """
+    fa_path = os.path.join(output_dir, upa, upa + ".fa")
+    meta_path = os.path.join(output_dir, upa, upa + ".meta")
+    if not os.path.exists(fa_path) or not os.path.exists(meta_path):
+        return False
+    try:
+        with open(meta_path, "r") as json_file:
+            data = json.load(json_file)
+    except:
+        return False
+    if set(data.keys()) != set(META_KEYS):
+        return False
+    return True
+
+
+def get_ip():
+    """
+    Get current ip address
+    """
+    ip = requests.get('https://ipv4.jsonip.com').json()['ip']
```

Comment:
This makes me uncomfortable... it seems like there should be a way to get it without an external network call. 

Good enough for now I suppose

Reply:
Updated!

Reply:
:+1: